src/battery-check/battery-check.c

```diff
@@ -0,0 +1,129 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <getopt.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "battery-util.h"
+#include "build.h"
+#include "constants.h"
+#include "errno-util.h"
+#include "glyph-util.h"
+#include "fd-util.h"
+#include "log.h"
+#include "main-func.h"
+#include "socket-util.h"
+#include "terminal-util.h"
+
+static void help(void) {
+        printf("%s\n\n"
+               "Checks battery level to see whether there's enough charge.\n\n"
+               "   -h --help            Show this help\n"
+               "      --version         Show package version\n",
+               program_invocation_short_name);
+}
+
+static void battery_check_send_plymouth_message(const char *message) {
+        assert(message);
+
+        int r;
+        ssize_t x, w;
+        static const union sockaddr_union sa = PLYMOUTH_SOCKET;
+        _cleanup_close_ int fd = -EBADF;
+        size_t n = strlen(message);
+
+        /* We set SOCK_NONBLOCK here so that we rather drop the
+         * message than wait for plymouth */
+        fd = socket(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
+        if (fd < 0) {
+                log_warning_errno(errno, "socket() failed: %m");
+                return;
+        }
+
+        r = connect(fd, &sa.sa, SOCKADDR_UN_LEN(sa.un));
+        if (r < 0) {
+                log_full_errno(IN_SET(errno, EAGAIN, ENOENT) || ERRNO_IS_DISCONNECT(errno) ? LOG_DEBUG : LOG_WARNING, r, "Connection to plymouth failed: %m");
```

Comment:
actually the second param log_full_errno() is supposed to be the error number. connect() pleaces the error in `errno`, hence the second param here should be `errno` rather than `r`

Reply:
> actually the second param log_full_errno() is supposed to be the error number. connect() pleaces the error in `errno`, hence the second param here should be `errno` rather than `r`

Awesome. Thank you sir.